src/components/side_panel/chart/chart_with_axis/design_panel.ts

```diff
@@ -76,6 +77,10 @@ export class ChartWithAxisDesignPanel<P extends Props = Props> extends Component
     return this.props.definition.dataSets.map((d, i) => d.label ?? `${ChartTerms.Series} ${i + 1}`);
   }
 
+  getPolynomialDegrees(): number[] {
+    return Array.from({ length: Math.min(10, this.getMaxPolynomialDegree()) }, (_, i) => i + 1);
```

Comment:
use the `range` helper which is much more readable IMO. Also the `Math.min(10` is already part of `getMaxPolynomialDegree`)
```suggestion
    return range(1, this.getMaxPolynomialDegree() +1);
```